mender-update/daemon/state_test.cpp

```diff
@@ -3207,6 +3209,28 @@ TEST_P(StateDeathTest, StateTransitionsTest) {
 	EXPECT_TRUE(mtesting::FileContains(status_log_path, content));
 }
 
+TEST(SignalHandlingTests, SigquitHadlingTest) {
+	mtesting::TemporaryDirectory tmpdir;
+	conf::MenderConfig config {
+		.data_store_dir = tmpdir.Path(),
+	};
+	context::MenderContext main_context {config};
+	auto err = main_context.Initialize();
+	mtesting::TestEventLoop event_loop {chrono::seconds {3}};
+	Context ctx {main_context, event_loop};
+
+	events::Timer signal_timer {event_loop};
+	signal_timer.AsyncWait(chrono::seconds {1}, [](error::Error err) { kill(getpid(), SIGQUIT); });
```

Comment:
Could use `raise` instead, which doesn't require `unistd.h` and `getpid`.

Reply:
Nice! I didn't know about `raise()`. However, it's man page says:
> If the signal causes a handler to be called, raise() will return only after the signal handler has returned
Not sure if that's what we want :)

Reply:
I expect that it will be fine, because the signal handler in this context is the async one in POSIX. While I don't know for sure about the Boost internals, almost certainly they are only using this handler to set some flag and trigger a condition somewhere, and then the real work is carried out later in the event loop.

Reply:
Well, it should be like that because the handler registration probably happens before the loop is started. Let's give it a try...